assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -61,45 +62,45 @@ const AttachmentContent = ( props ) => {
 		const { getSettings } = select( 'amp/story' );
 		const { allowedPageAttachmentPostTypes } = getSettings();
 		return allowedPageAttachmentPostTypes;
-	} );
+	}, [] );
 
-	const fetchSelectedPost = () => {
+	const fetchSelectedPost = useCallback( () => {
 		const restBase = allowedPostTypes[ postType ] || `${ postType }s`;
 
-		isStillMounted = true;
+		isStillMounted.current = true;
 		if ( postId ) {
 			setIsFetching( true );
-			const currentFetchRequest = fetchRequest = apiFetch( {
+			const currentFetchRequest = fetchRequest.current = apiFetch( {
 				path: `/wp/v2/${ restBase }/${ postId }`,
 			} ).then(
 				( post ) => {
-					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+					if ( isStillMounted.current && fetchRequest.current === currentFetchRequest ) {
 						setSelectedPost( post );
 						setFailedToFetch( false );
 						setIsFetching( false );
 					}
 				}
 			).catch(
 				() => {
-					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+					if ( isStillMounted.current && fetchRequest.current === currentFetchRequest ) {
 						setSelectedPost( null );
 						setFailedToFetch( true );
 						setIsFetching( false );
 					}
 				}
 			);
 		}
-	};
+	}, [ postId, postType, allowedPostTypes ] );
 
 	useEffect( () => {
 		return () => {
-			isStillMounted = false;
+			isStillMounted.current = false;
 		};
 	}, [] );
 
 	useEffect( () => {
 		fetchSelectedPost();
-	}, [ postId ] );
+	}, [ fetchSelectedPost ] );
```

Comment:
^ Why this change? Shouldn't we track the change of `postId` here instead?

Reply:
Functions that are used within an effect should be listed as dependencies, see https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

`fetchSelectedPost` itself is wrapped in `useCallback` which has `postId` as a dependency, so this is never out of sync

Reply:
Okay, good to know! 👍